### ISSUE TITLE
Google Tag Manager nicht eingebunden

### ISSUE THREAD

**Issue by benediktrupp** [heimrichhannot/contao-google-tag-manager-bundle#3]
Hallo zusammen,

wir sind diese drei Schritte nach Anleitung durchgegangen und haben das im Cantao Backend implementiert.

**Set your google tag manager container id under the contao backend settings.
Copy the following code snippet and paste it as close to the opening <head> tag as possible on every page of your website. <?= $this->googleTagManagerHead ?>
Copy the following code snippet and paste it as close to the opening <body> tag as possible on every page of your website. <?= $this->googleTagManagerBody ?>**

Allerdings ist nach dem Bearbeiten der drei oberen Punkte der Google Tag Manager weiterhin nicht eingebunden.
Wir finden hier leider das Problem nicht.

Müssen die beiden unteren Punkte noch beachtet werden und wenn ja, wie können diese korrekt integriert werden:

**add data layer variables to a page

System::getContainer()->get('huh.google_tag_manager.data_layer')->setContent(['key' => 'value']);
add an a tag with data layer event handler

System::getContainer()->get('huh.google_tag_manager.tag_manager')->getLink(class, link, linkText, ['event': 'button1-click'])
This will output: <a href="link" class="class" onclick="dataLayer.push({'event': 'button1-click'});">linkText</a>**

Vielleicht kann uns jemand bei der Suche nach dem Problem helfen, damit wir den GTM korrekt einbinden können.

Über jede Hilfe wäre ich froh!

Vielen Dank!

**Comment by koertho** [comment]
Wie und wo habt ihr `<?= $this->googleTagManagerHead ?>` und `<?= $this->googleTagManagerBody ?>` eingebunden? Der Code gehört in da Page-Template (fe_page.html5)

**Comment by GMTeams** [comment]
I also tried to add GTM to my Contao 4.13, but I can't see any Google GTM source code in my HTML. 
Steps which I did:

- installed contao-google-tag-manager-bundle 1.2.0
- updated database
- added the code to fe_page.html5
- added container id in settings
I also tried it without being logge in in the same browser session in contao.

Maybe I'm getting something wrong here.
What I would expect is that a code like the following should appear in my HTML code:
```
<!-- Google Tag Manager -->

<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':

new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],

j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=

'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);

})(window,document,'script','dataLayer','GTM-xxxxxx');</script>

<!-- End Google Tag Manager -->
```

My goal is to use the advanced features.
Thank you very much.

**Comment by benediktrupp** [comment]
> Wie und wo habt ihr `<?= $this->googleTagManagerHead ?>` und `<?= $this->googleTagManagerBody ?>` eingebunden? Der Code gehört in da Page-Template (fe_page.html5)

Ja, genau dort haben wir es eingebunden. Hast du sonst eine Idee, die uns helfen könnte oder Lust und Zeit auf einen Austausch?

**Comment by koertho** [comment]
Wir nutzen das Bundle selbst nicht mehr, daher ist es schwierig für mich, da gerade Zeit reinzustecken. Eventuell kann einer von euch mal schauen und gerne auch nen PR bereit stellen. 
Ansonsten sollte für den Google Tag-manager eigentlich auch das von Contao mitgelieferte Google Analytics Template reichen. Weitere Funktionen können ja dann per JS anstatt PHP hinzugefügt werden, von daher braucht es dieses Bundle hier nicht wirklich.
